contracts/Recruitment.sol

```diff
@@ -72,7 +84,143 @@ contract Recruitment {
+    uint256 jobId = jobIdCounter.current();
+    FrontDoorStructs.Job memory job = FrontDoorStructs.Job(jobId, bounty, maxSalary, minSalary, companyName, location, role, jobURL);
+    jobList[jobId] = job;
+    jobIdCounter.increment();
+  }
+
+  /**
+    * @notice Getter function of Job.
+    * @param jobId The ID of a job to fetch from the mapping
+    * @return Job The type of FrontDoorStruct.Job 
+  */
+  function getJob(uint256 jobId) external view returns(FrontDoorStructs.Job memory){
+    return jobList[jobId];
+  }
+
+  /**
+    * @notice Registers a referral
+    * @param jobId The job ID already registered with the contract.
+    * @param refereeWallet The address of referee.
+    * @param refereeMail The email of the referee.
+  */
+  function registerReferral(uint256 jobId, address refereeWallet, string memory refereeMail) external {
+    FrontDoorStructs.Referee memory referee = refereeList[refereeWallet];
+    FrontDoorStructs.Referrer memory referrer = referrerList[msg.sender];
+    FrontDoorStructs.Job memory job = jobList[jobId];
+
+    if (referee.wallet == address(0)) {
+      referee = FrontDoorStructs.Referee(refereeWallet, refereeMail, 0);
+      refereeList[refereeWallet] = referee;
+    }
+
+    FrontDoorStructs.Referral memory referral = FrontDoorStructs.Referral(referralCounter.current(), false, referrer, referee, job);
+
+    referralIndex[msg.sender].push(referralCounter.current());
+    referralList[referralCounter.current()] = referral;
+    referralCounter.increment();
+
+  }
+
+  /**
+    * @notice Confirmation of a referral by the msg.sender (if the msg.sender is the referee)
+    * @param refId The referral ID already recorded in the smart contract.
+  */
+  function confirmReferral(uint256 refId) external {
+    
+    FrontDoorStructs.Referral memory referral = referralList[refId];
+
+    if (referral.referee.wallet != msg.sender) revert Errors.SenderIsNotReferee(); // The msg.sender should be the referee.
+
+    referral.confirmed = true;
+    referralList[refId] = referral;
+  }
+
+  /**
+    * @notice Getter function of Referral IDs of the msg.sender.
+    * @return uint256[] An array of uint256 with all the referral ids of msg.sender
+  */
+  function getReferralIDs() external view returns(uint256[] memory){
+    return referralIndex[msg.sender];
+  }
+
+  /**
+    * @notice Getter function of Referral.
+    * @param refId The referral ID already recorded in the smart contract.
+    * @return Referral The type of FrontDoorStruct.Referral
+  */
+  function getReferral(uint256 refId) external view returns(FrontDoorStructs.Referral memory) {
+    return referralList[refId];
+  }
+
+
   // function withdrawTokens(bytes32 symbol, uint256 amount) external {
   //   // to be continued...
   //   require(msg.sender == owner, 'Only owner can withdraw!');
```

Comment:
A last thing but this is just if you have time and feel like it. There is a tests file where I am adding some tests written using chai package. If you don't know it just check the other examples at test/recruitment.js. So the idea is if you can add 2 tests one to register a referral another to confirm a referral. As an expected result on these 2 tests you can read the referrals and expect that in the first you have a registered non confirmed referral and in the last that you have a confirmed referral. Thank you so much :)

Reply:
My apologies, I haven't worked with Chai so far, it may take time to get it done.